### ISSUE TITLE
[no-shadow] false positive with enum

### ISSUE THREAD

**Issue by lmichelin** [typescript-eslint/typescript-eslint#2629]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

Isolated reproduction repository: https://github.com/lmichelin/typescript-eslint-no-shadow-bug

Github Action showing the issue: https://github.com/lmichelin/typescript-eslint-no-shadow-bug/runs/1205468699

```JSON
{
  "root": true,
  "parser": "@typescript-eslint/parser",
  "parserOptions": { "project": ["./tsconfig.json"] },
  "extends": ["plugin:@typescript-eslint/recommended"],
  "rules": {
    "no-shadow": "error"
  }
}
```

```TS
export enum MyEnum {
  a = "a",
  b = "b",
}
```

```JSON
{
  "compilerOptions": {
    "strict": true,
    "target": "ES6"
  }
}
```

**Expected Result**

No error reported by eslint.

**Actual Result**

`1:13  error  'MyEnum' is already declared in the upper scope  no-shadow`

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.3.0` |
| `@typescript-eslint/parser`        | `4.3.0` |
| `TypeScript`                       | `4.0.3` |
| `ESLint`                           | `7.10.0` |
| `node`                             | `12.18.4` |

**Comment by yeonjuan** [comment]
@lmichelin See [FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md#i-am-using-a-rule-from-eslint-core-and-it-doesnt-work-correctly-with-typescript-code). you should off the base rule (`no-shadow`) because it cannot work properly with ts syntax. you should turn on @typescript-eslint rule

```
  "rules": {
    "no-shadow": "off",
    "@typescript-eslint/no-shadow": "off",
  }
```

**Comment by lmichelin** [comment]
Actually I opened this issue because `no-shadow` seemed to work properly with @typescript-eslint v3.10.1, the problem appeared with v4, but if it's recommended to use `@typescript-eslint/no-shadow`I will use it, thanks :)

**Comment by bradzacher** [comment]
Please read the v4 release notes.

Major releases have breaking changes and we explain it there.

**Comment by lmichelin** [comment]
it's very clear, thanks 😉

**Comment by mikemaccana** [comment]
It looks like this was included in *non-breaking* in [v4 release notes](https://github.com/typescript-eslint/typescript-eslint/releases/tag/v4.0.0)? but is actually breaking according to https://github.com/typescript-eslint/typescript-eslint/pull/2039

**Comment by bradzacher** [comment]
@mikemaccana - the addition of a new lint rule is not breaking.
The addition of the new lint rule did not break the core rule. They are two completely separate rules.

Regardless - the existing lint rule was already broken, so it just got broken further.
And that breakage is listed under breaking - the PR you mentioned and the big section titled "TypeScript Scope Analysis", which also lists the rule you need to start using.
